### PR TITLE
Integration objects tenant config service

### PR DESCRIPTION
**Pull request by vidhyakpai**
Updated the calls to the generic methods validateLength and validatePattern
Updated test cases
cleaned up the extension files

### CODE REVIEW

File: packages/integration-objects/srv/validations/ValidatePrimaryKeysForTenantConfig.js [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
-const { logger } = require('../logging/Logger');
-const ErrorStatuses = require('../utils/ErrorStatuses');
-const TenantConfigExtensionConstants = require('../utils/TenantConfigExtensionConstants');
+const { validateLength } = require('./ValidateLength');
+const { validatePattern } = require('./ValidatePattern');
+const destinationNameField = 'destinationName';
+const defaultDataSourceField = 'defaultDataSource';
+const defaultDataProviderCodeField = 'defaultDataProviderCode';
+const MAX_DESTINATION_NAME_LENGTH = 200;
+const MAX_VALUE_DATA_SOURCE_DATA_PROVIDER = 15;
```
[the remark this answers is not in the document]
Comment (vidhyakpai) [reply packages/integration-objects/srv/validations/ValidatePrimaryKeysForTenantConfig.js:10]: updated these in the ticket.

File: packages/integration-objects/srv/validations/ValidatePrimaryKeysForTenantConfig.js [hunk 2]
```diff
@@ -1,29 +1,29 @@
...
-      throw new ValidationError(
-        TenantConfigExtensionConstants.INVALID_DESTINATION_NAME_FIELD_VALUE,
-        ErrorStatuses.BAD_REQUEST
-      );
-    }
+    validatePattern(
+      Constants.DESTINATION_NAME_PATTERN,
+      destinationNameField,
+      data.destinationName,
+      MAX_DESTINATION_NAME_LENGTH
```
Comment (FrankEssenberger) [inline packages/integration-objects/srv/validations/ValidatePrimaryKeysForTenantConfig.js:25]: I had this point already on other PRs. I would really recommend to make the `validatePattern` generic in a sense that it tests a pattern for a given value and pattern. It uses then the field name and pattern info to make a nice error message. If your message is about the length than pass it in. But if the method stays like this it is not really usable in other places.

File: packages/integration-objects/test/ValidatePrimaryKeysForTenantConfig.test.js [hunk 3]
```diff
@@ -13,98 +10,91 @@ describe('validate fields for tenant config', function () {
...
-    await expect(
-      validatePrimaryCompositeKeysForTenantConfig(reqObj)
-    ).rejects.toThrow(
-      new ValidationError(
-        TenantConfigExtensionConstants.INVALID_DEFAULT_DATA_PROVIDER_FIELD_VALUE,
-        ErrorStatuses.BAD_REQUEST
-      )
-    );
+    try {
+      validatePrimaryCompositeKeysForTenantConfig(reqObj);
```
Comment (FrankEssenberger) [inline packages/integration-objects/test/ValidatePrimaryKeysForTenantConfig.test.js:23]: same here for simple exception checks use the jest build in mehtods like expect().throws etc.

**Review by FrankEssenberger — CHANGES_REQUESTED** [review]
Please consider my comments and also merge main so that the test are executed on the PRs.

File: packages/integration-objects/test/ValidatePrimaryKeysForTenantConfig.test.js (hunk 3 above)
Comment (vidhyakpai) [reply packages/integration-objects/test/ValidatePrimaryKeysForTenantConfig.test.js]: this is done.

File: packages/integration-objects/test/ValidatePrimaryKeysForTenantConfig.test.js [hunk 4]
```diff
@@ -98,13 +75,8 @@ describe('validate fields for tenant config', function () {
...
       destinationName: '*ABc'
     };
-    await expect(
-      validatePrimaryCompositeKeysForTenantConfig(reqObj)
-    ).rejects.toThrow(
-      new ValidationError(
-        TenantConfigExtensionConstants.INVALID_DESTINATION_NAME_FIELD_VALUE,
-        ErrorStatuses.BAD_REQUEST
-      )
+    await expect(validatePrimaryCompositeKeysForTenantConfig(reqObj)).rejects.toThrow(
```
[the remark this answers is not in the document]
Comment (vidhyakpai) [reply packages/integration-objects/test/ValidatePrimaryKeysForTenantConfig.test.js]: Done

File: packages/integration-objects/srv/validations/ValidatePrimaryKeysForTenantConfig.js (hunk 2 above)
Comment (vidhyakpai) [reply packages/integration-objects/srv/validations/ValidatePrimaryKeysForTenantConfig.js:25]: All the errors resulting from the pattern check have details about the max length. So I'd like to keep the errors the same as that in the Java code. However, I have noted this and updated it in my issues and will relook at it in the next release.

File: packages/integration-objects/srv/integration-objects/simple-tenant-configuration-for-conversion-service.js [hunk 5]
```diff
@@ -85,21 +84,20 @@ module.exports = srv => {
...
 
-    let currentValuesIsNotEmpty = true;
-    if (Constants.CREATE_EVENT !== req.event) {
-      currentValues = await fetchCurrentAttributeValues(req, req.user.tenant, TenantConfigForConversions);
-      currentValuesIsNotEmpty = !util.isNullish(currentValues);
-    }
-
-    if (currentValuesIsNotEmpty) {
-      await writeAuditLog(req, auditParams, currentValues, type);
+    if (req.event === Constants.CREATE_EVENT) {
```
Comment (marikaner) [inline packages/integration-objects/srv/integration-objects/simple-tenant-configuration-for-conversion-service.js:87]: ???? This is the exact same code as in the other service. Please avoid code duplication and reuse existing functionality.

**Review by marikaner — APPROVED** [review]
As discussed, LGTM

**Pull request merged**